pkg/kgo/compression.go

```diff
@@ -137,6 +137,7 @@ out:
 			}
 			w.Close()
 			c.lz4Pool = sync.Pool{New: fn}
+		case codecSnappy:
```

Comment:
I'd prefer this be above lz4, since it's 2 (0 none, 1 gzip, 2 snappy)

But, this case doesn't exist currently because it's an unneeded case -- there is no pool no initialize. I'd either drop the case, or add a one line comment for why the case exists: `// no pool needed for snappy`

Reply:
Sure, I can add a comment. I added this line to satisfy your linter - it was [lamenting](https://github.com/twmb/franz-go/actions/runs/3595872490) `missing cases in switch of type codecType: codecSnappy (exhaustive)`.

Reply:
interesting, perhaps that linter is triggering now because this is a dedicated type